include/lib/stdlib/machine/_types.h

```diff
@@ -110,8 +73,23 @@ typedef	__int8_t	__int_least8_t;
 typedef	__int16_t	__int_least16_t;
 typedef	__int32_t	__int_least32_t;
 typedef	__int64_t	__int_least64_t;
+typedef	long		__ptrdiff_t;		/* ptr1 - ptr2 */
+typedef	long		__register_t;
+typedef	long		__segsz_t;		/* segment size (in pages) */
+/* differentiate for "%z" in printf() */
+#ifdef	AARCH32
+typedef	unsigned int	__size_t;		/* sizeof() */
+typedef	int		__ssize_t;		/* byte count or error */
```

Comment:
Perhaps we can fix tf_printf to always using long for %z rather than have the #ifdef here?

Reply:
Yes, good point.